src/almo_scf_optimizer.F

```diff
@@ -5678,24 +5486,34 @@ SUBROUTINE compute_preconditioner(domain_prec_out, m_prec_out, m_ks, m_s, &
       ! invert using various algorithms
       IF (special_case .EQ. xalmo_case_block_diag) THEN ! non-overlapping diagonal blocks
 
-         CALL pseudo_invert_diagonal_blk( &
-            matrix_in=m_tmp_nn_1, &
-            matrix_out=m_prec_out, &
-            nocc=nocc_of_domain(:) &
-            )
+         IF (skip_inversion) THEN
+            CPABORT("NYI: impose blk structure on m_prec_out first")
+            CALL dbcsr_copy(m_prec_out, m_tmp_nn_1, keep_sparsity=.TRUE.)
```

Comment:
There is little point in calling another function after a `CPABORT`.
If this is needed once the function gets implemented then its fine, but you might want to comment it out like other cases, otherwise it looks like a misplaced statement.

Reply:
This block is being implemented.